### PR TITLE
Add test cases for port auto negotiation feature

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
### Description of PR

Summary:
Add test cases for port auto negotiation feature

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?

Add test cases for port auto negotiation feature

#### How did you do it?

1. Add test case to verify "config interface advertised-speeds <interface_name> all"
2. Add test case to verify "config interface advertised-speeds <interface_name> <speed>"
3. Add test case to verify "config interface speed <interface_name> <speed>"

#### How did you verify/test it?

Manually run the test case on Mellanox platform

#### Any platform specific information?

N/A

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

**Comment by Junchao-Mellanox** [comment]
The test failure is not related to the change.

**Review by jleveque — APPROVED** [review]

**Review by jleveque — APPROVED** [review]

File: tests/common/devices/sonic.py [hunk 1]
```diff
@@ -1361,6 +1361,83 @@ def get_up_ip_ports(self):
...
+        Args:
+            interface_name (str): Interface name
+
+        Returns:
+            str: SONiC style interface speed value. E.g, 1G=1000, 10G=10000, 100G=100000.
+        """
+        cmd = 'sonic-db-cli APPL_DB HGET \"PORT_TABLE:{}\" \"{}\"'.format(interface_name, 'speed')
+        speed = self.shell(cmd)['stdout'].strip()
+        return speed
     def get_rsyslog_ipv4(self):
```
Comment (liat-grozovik) [inline tests/common/devices/sonic.py:1441]: please add new line

File: tests/platform_tests/test_auto_negotiation.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        if expect_speed:
+            int_status = show_interface_output['int_status']
+            for dut_port in dut_ports:
+                actual_speed = int_status[dut_port]['speed'][:-1] + '000'
+                if actual_speed != expect_speed:
+                    return False
+        return True
+    else:
+        return False
+
```
Comment (liat-grozovik) [inline tests/platform_tests/test_auto_negotiation.py:153]: remove unnecessary new lines

File: tests/platform_tests/test_auto_negotiation.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        logger.info('Get supported speeds for {} {}: {}'.format(duthost, dut_port_name, output))
+        if not output:
+            return None
+        pos = output.rfind('(')
+        if pos == -1:
+            return None
+        speeds_str = output[pos+1:-1]
+        speeds =  [speed[:-1] + '000' for speed in speeds_str.split(',')]
+        cls.supported_speeds[(duthost, dut_port_name)] = speeds
+        return speeds
```
Comment (liat-grozovik) [inline tests/platform_tests/test_auto_negotiation.py:358]: add new line

File: tests/platform_tests/test_auto_negotiation.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
+import logging
```
Comment (liat-grozovik) [inline tests/platform_tests/test_auto_negotiation.py:1]: Suggest to add a header to this file with a summary of what is the motivation and what will be tested.
If you have an HLD please refer to it as well. if not, just short description

**Review by liat-grozovik — COMMENTED** [review]
LGTM some minor cosmetic changes required

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**